entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/config/DataSourceConfig.java

```diff
@@ -64,6 +78,43 @@ public MongoClient mongoClient() {
         (ConnectionPoolSettings.Builder builder) ->
             builder.maxConnectionIdleTime(mongoMaxIdleTimeMillisec, TimeUnit.MILLISECONDS);
 
+    try {
+      // build SSL context that uses separate truststore (needs to be copied and specified via
+      // mongo.truststore property)
+      if (!truststorePath.isBlank()) {
+        File tsFile =
+            ResourceUtils.getFile("classpath:" + truststorePath); // throws error if not available
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+        ks.load(new FileInputStream(tsFile), truststorePwd.toCharArray());
+        logger.info("Read truststore file {}", truststorePath);
+
+        TrustManagerFactory trustFactory =
+            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init(ks);
+
+        SSLContext sslContext = SSLContext.getInstance("TLSv1.3");
```

Comment:
probably we should put the TLS version in the properties, not sure if it is required now, but Mongo switched to TLS 2, and uses TLS1 as legacy support according to the documentation

Reply:
that is probably also fine, I would just avoid using allways the hardcoded value.

Reply:
and indeed, I made a confusion between 1.2 and 2.0 ... https://www.mongodb.com/docs/manual/tutorial/configure-ssl/